client/src/pages/SignUpModal.tsx

```diff
@@ -1,151 +1,196 @@
+  const [redirect, setRedirect] = useState("");
+  const requirementsAreFulfilled =
+    !hasOneLowerCase ||
+    !hasOneUpperCase ||
+    !hasOneNumber ||
+    !hasOneSymbol ||
+    !hasTwelveCharacterMin;
 
-    const handleClick = async (e: React.FormEvent<HTMLFormElement>) => { 
-        e.preventDefault();
-        createNewAccount(email, password);
-        setRedirectToHome(true);
-    };
+  const handleClick = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    createNewAccount(email, password)
+      .then((result) => {
+        setErrors(result);
+        if (result.email === "" && result.password === "") {
+          setRedirect("/"); //TODO: change this to admin view endpoint
+        }
+      })
+      .catch((err) => {
+        setErrors(err);
+      });
+  };
 
-    const onChangeEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
-        setEmail(e.target.value);
-    };
-     
-    const requirementToTestMap = new Map ([
-        ['lowerCase', /^(?=.*[a-z])/],
-        ['upperCase', /^(?=.*[A-Z])/],
-        ['number', /^(?=.*[0-9])/],
-        ['symbol', /^(?=.*[*!@#$%^&(){}[\]:;<>,.?/~_+\-=|\\])/],
-        ['twelveCharacters', /^(?=.{12,})/]
-    ]);
-        
-    const requirementToStateSetterMap = new Map ([
-        ['lowerCase', setHasOneLowerCase],
-        ['upperCase', setHasOneUpperCase],
-        ['number', setHasOneNumber],
-        ['symbol', setHasOneSymbol],
-        ['twelveCharacters', setHasTwelveCharacterMin],
-    ]);
-        
-    const requirementToMessageMap = new Map ([
-        ['lowerCase', "at least 1 lowercase letter"],
-        ['upperCase', "at least 1 uppercase letter"],
-        ['number', "at least 1 number"],
-        ['symbol', 'at least 1 symbol'],
-        ['twelveCharacters', '12 characters minimum'],
-    ]);
-          
-          
-    const onChangePass = (e: React.ChangeEvent<HTMLInputElement>) => {
-        const password = e.target.value;
-        const copy: string[] = requirements;
-        
-        requirementToTestMap.forEach((test, key) => {
-            const result = test.test(password)!; 
-            const message : string = requirementToMessageMap.get(key)!;
-            requirementToStateSetterMap.get(key)!(result)
-            
-            // if requirement not in string and not in array, push into array
-            if(!result && !copy.includes(message)){
-                copy.push(message);
-            }
-            // if requirement is in the string and the array has it, remove it from array
-            else if(result && copy.includes(message)){
-                const index = copy.indexOf(message);
-                copy.splice(index, 1); 
-            }
-            setRequirements(copy);
-        });
+  const onChangeEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+  };
 
-        setPassword(password);
-    }
+  const requirementToTestMap = new Map([
```

Comment:
Would it be possible to put file reformatting in another PR? Lol it's hard to understand these diffs

Reply:
this one was pretty tough lol but i think the diffs are more broken up and digestible now

Reply:
OMG SORRY Y'ALL this is so late holy here's the styling feedback
- I think the modal header font size should be bigger? is it 24px?
- are we doing the icon symbols :O ? I noticed that the alert icon wasn't there
- left margin of the filler text in the sign in is a bit too big I think? It should be the same as the margin in "Modal Input Field Errors"